src/lazy/binary/raw/v1_1/reader.rs

```diff
@@ -457,4 +457,81 @@ mod tests {
 
         Ok(())
     }
+
+    #[test]
+    fn structs() -> IonResult<()> {
+        use crate::lazy::decoder::{LazyRawFieldExpr, LazyRawFieldName};
+        use crate::raw_symbol_token_ref::RawSymbolTokenRef;
+
+        #[rustfmt::skip]
+        #[allow(clippy::type_complexity)]
+        let tests: &[(&[u8], &[(RawSymbolTokenRef, IonType)])] = &[
+            // Symbol Address
+            (
+                // { $10: 1, $11: 2 }
+                &[0xC6, 0x15, 0x51, 0x01, 0x17, 0x51, 0x02],
+                &[
+                    (10usize.into(), IonType::Int),
+                    (11usize.into(), IonType::Int),
+                ]
+            ),
+            (
+                // { $10: '', $11: 0e0 }
+                &[0xC4, 0x15, 0x90, 0x17, 0x5A],
+                &[
+                    (10usize.into(), IonType::Symbol),
+                    (11usize.into(), IonType::Float),
+                ],
+            ),
+            (
+                // { $10: <NOP>, $11: 0e0 } - with nops, skip the NOP'd fields.
+                &[ 0xC4, 0x15, 0xEC, 0x17, 0x5A ],
+                &[
+                    (11usize.into(), IonType::Float),
+                ],
+            ),
+            (
+                // { $10: "variable length struct" }
+                &[
+                    0xFC, 0x33, 0x15, 0xF8, 0x2D, 0x76, 0x61, 0x72, 0x69, 0x61,
+                    0x62, 0x6C, 0x65, 0x20, 0x6c, 0x65, 0x6E, 0x67, 0x74, 0x68,
+                    0x20, 0x73, 0x74, 0x72, 0x75, 0x63, 0x74
+                ],
+                &[ (10usize.into(), IonType::String) ],
+            ),
+            // FlexSym
+            (
+                // { "foo": 1, $11: 2 }
+                &[ 0xD9, 0xFB, 0x66, 0x6F, 0x6F, 0x51, 0x01, 0x17, 0x91, 0x02],
+                &[ ("foo".into(), IonType::Int), (11usize.into(), IonType::Symbol)],
```

Comment:
I realize you're implementing an older copy of the spec, but the most recent version merged the distinct struct "kinds" into a single kind that can switch from symbol addresses to flexsym partway through a struct by emitting a `FlexUInt 0` in field name position. This can wait for a follow-on PR since we have lots of updates to make around opcodes.